api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go

```diff
@@ -210,8 +211,14 @@ func (i *MaterializeStageProvider) Process(ctx context.Context, mgrContext conte
 		}
 	}
 
+	// put instance type catalogs at the end since they may depend on solution and target
+	sort.Slice(prefixedNames, func(i, j int) bool {
```

Comment:
This still changes our previous behavior. Is it better we respect customers' input?

Reply:
I think our previous behavior is not good enough. We should not throw error if instance, solution, target are materialized in the same batch.

Reply:
We also have catalogs which will have order check per parent name. I prefer we always respect users' input order. 
Even if users use list providers to list the catalogs, users can write expressions to select the catalogs and ensure the correct order then pass to materialize stage provider. It is hard for us to assume what is the correct order in materialize stage provider.